### PR TITLE
New DSC Alarm Action

### PR DESCRIPTION
**Pull request by RSStephens**
This is a new implementation for a DSC Alarm Action to add the ability to send DSC Alarm commands through the rule subsystem.  This was a request made from the openHAB community forum as seen here: https://community.openhab.org/t/dsc-binding-how-to-send-a-command-string-similar-to-keypad/2708/9.

### CODE REVIEW

**Comment by RSStephens** [comment]
Thanks @mrguessed.  The errors are fixed.  I appreciate it.

File: bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -30,7 +30,12 @@ Import-Package: gnu.io,
  org.slf4j
 Service-Component: OSGI-INF/activebinding.xml, OSGI-INF/genericbindingprovider.xml
 Bundle-ClassPath: .
-Export-Package: org.openhab.binding.dscalarm
+Export-Package: org.openhab.binding.dscalarm;
+  uses:="org.openhab.binding.dscalarm.internal,
+   org.openhab.core.items,
+   org.openhab.core.binding,
+   org.openhab.binding.dscalarm.internal.model"
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF]: please revert these additions

File: bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -4,7 +4,7 @@ Bundle-SymbolicName: org.openhab.binding.dscalarm
 Bundle-Version: 1.8.0.qualifier
 Bundle-Activator: org.openhab.binding.dscalarm.internal.DSCAlarmActivator
 Bundle-Vendor: openHAB.org
-Bundle-RequiredExecutionEnvironment: J2SE-1.5
+Bundle-RequiredExecutionEnvironment: JavaSE-1.6
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF]: any requirement to change to 1.6?

**Comment by teichsta** [comment]
Thanks @RSStephens for this contribution (and @mrguessed for the first review). Overall this PR lgtm however i've also found some small issues. Could you finally please squash your commits into one? Thanks …

File: bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF (hunk 2 above)
Comment (RSStephens) [inline bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF]: I was getting a "must override a superclass method" error after adding the new method to the binding, so I bumped the execution environment up to 1.6.

**Comment by RSStephens** [comment]
Hello Thomas,

I made the changes you requested and squashed the commits.  Thank you.

File: bundles/action/org.openhab.action.dscalarm/.project [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>org.openhab.action.dscalarm</name>
+	<comment>This is the ${binding-name} binding of the open Home Automation Bus (openHAB)</comment>
```
Comment (watou) [inline bundles/action/org.openhab.action.dscalarm/.project]: Very minor: it's the *action*, not the *binding*.

File: bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -30,7 +30,9 @@ Import-Package: gnu.io,
  org.slf4j
 Service-Component: OSGI-INF/activebinding.xml, OSGI-INF/genericbindingprovider.xml
 Bundle-ClassPath: .
-Export-Package: org.openhab.binding.dscalarm
+Export-Package: org.openhab.binding.dscalarm,
+ org.openhab.binding.dscalarm.internal
+Bundle-ActivationPolicy: lazy
```
Comment (watou) [inline bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF]: Repeating what Kai said, you must not export internal packages.  In the Ecobee binding, I renamed my "internal.messages" package to "messages" so at least it's not considered internal.

**Comment by watou** [comment]
Looking at the last file Thomas changed for #2967, it looks like the file `targetplatform/openHAB Designer (Mac).launch` also needs to be updated for your actions to be visible in the Designer.

File: bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF (hunk 4 above)
Comment (RSStephens) [inline bundles/binding/org.openhab.binding.dscalarm/META-INF/MANIFEST.MF]: I don't know why I am exporting .internal anyway. I don't think it is necessary.  I will change that.  Thanks @watou.

**Comment by teichsta** [comment]
Thanks again @RSStephens and @watou for the thorough review!

**Comment by watou** [comment]
@RSStephens , could you update the [Actions wiki page](https://github.com/openhab/openhab/wiki/Actions) with either the documentation for this new bundle, or a link to a page dedicated to the new bundle?  Also, consider adding a note and link to the [binding wiki page](https://github.com/openhab/openhab/wiki/DSC-Alarm-Binding) to explain the new functionality that DSC alarm panel binding users could have if they added in the new action bundle.  Make sure to note the availability of the action bundle in openHAB 1.8 and later releases.  Thank you!

**Comment by RSStephens** [comment]
Yes I will update the wiki.  Thanks.

**Pull request merged**